src/docs/development/packages-and-plugins/plugin-api-migration.md

```diff
@@ -74,27 +74,52 @@ The following instructions outline the steps for supporting the new API:
       // now automatically registers plugins.
     }
     ```
+    
+1. (Optional) If you removed `MainActivity.java`, update the `<plugin_name>/example/android/app/src/main/AndroidManifest.xml` to use `io.flutter.embedding.android.FlutterActivity`. For example:
+
+    <?code-excerpt "AndroidManifest.xml" title?>
+    ```xml
+     <activity android:name="io.flutter.embedding.android.FlutterActivity"
+            android:theme="@style/LaunchTheme"
+   android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale"
+            android:hardwareAccelerated="true"
+            android:windowSoftInputMode="adjustResize">
+            <meta-data
+                android:name="io.flutter.app.android.SplashScreenUntilFirstFrame"
+                android:value="true" />
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+        </activity>
+    ```
 
 1. (Optional) Create an `EmbeddingV1Activity.java` file that uses the v1
    embedding for the example project in the same folder as `MainActivity` to
-   keep testing the v1 embedding's compatibility with your plugin. For example:
+   keep testing the v1 embedding's compatibility with your plugin. Note that
+   you have to manually register all the plugins instead of using `GeneratedPluginRegistrant`.
+   For example:
 
     <?code-excerpt "EmbeddingV1Activity.java" title?>
     ```java
-    package io.flutter.plugins.firebasecoreexample;
+    package io.flutter.plugins.batteryexample;
 
     import android.os.Bundle;
+    import dev.flutter.plugins.e2e.E2EPlugin;
     import io.flutter.app.FlutterActivity;
-    import io.flutter.plugins.GeneratedPluginRegistrant;
+    import io.flutter.plugins.battery.BatteryPlugin;
 
-    public class EmbeddingV1Activity extends FlutterActivity {
+    public class EmbedderV1Activity extends FlutterActivity {
```

Comment:
pedantic nit: We generally use "embedder" to refer to `embedder.h` in the engine, and "embedding" to refer to the actual Android embedding.

```suggestion
    public class EmbeddingV1Activity extends FlutterActivity {
```